### PR TITLE
set correct min cmake version.. fix print compile msg. remove old boost references

### PR DESCRIPTION
**Pull request by mosfet80**
set correct min. cmake version 3.11

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7041">pipeline status</a> for feature branch [PR_7041](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7041). Pipeline [566591437](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/566591437) was triggered at [f60deb6](https://github.com/FreeCAD/FreeCAD/pull/7041/commits/f60deb6361db0a81dfad16ba2e3cd6fbb42a2d29). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # March 2021, the oldest we have to support : Ununtu 18.04 LTS
-cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.11 FATAL_ERROR)
```
Comment (donovaly) [inline CMakeLists.txt:2]: For FreeCAD 1.0 we require CMake 3.16, see https://wiki.freecadweb.org/FreeCAD_1.0_Development_Cycle

File: cMake/FreeCAD_Helpers/CompilerChecksAndSetups.cmake [hunk 2]
```diff
@@ -9,19 +9,7 @@ macro(CompilerChecksAndSetups)
...
 
-    # Needed for boost1.69
-    # Avoid that Python (pyerrors.h) defines snprintf and vsnprintf
-    if (MSVC AND NOT MSVC_VERSION VERSION_LESS 1900)
-        add_definitions(-DHAVE_SNPRINTF)
-    elseif (MINGW)
-        add_definitions(-DHAVE_SNPRINTF)
-    endif()
-
-    # Allow developers to use Boost < 1.55
```
Comment (donovaly) [inline cMake/FreeCAD_Helpers/CompilerChecksAndSetups.cmake:20]: Please don't change too many different things. This PR should be to address CMake.
Please make a separate PR for boost.

File: cMake/FreeCAD_Helpers/CompilerChecksAndSetups.cmake [hunk 3]
```diff
@@ -46,10 +34,6 @@ macro(CompilerChecksAndSetups)
         set(CMAKE_CXX_STANDARD 20)
     elseif(${BUILD_ENABLE_CXX_STD} MATCHES "C\\+\\+17")
         set(CMAKE_CXX_STANDARD 17)
-    elseif(${BUILD_ENABLE_CXX_STD} MATCHES "C\\+\\+14")
```
Comment (donovaly) [inline cMake/FreeCAD_Helpers/CompilerChecksAndSetups.cmake:49]: same here, please a separate PR for C++

**Review by donovaly — CHANGES_REQUESTED** [review]
I propose to close this PR and make 3 new PRs:
- one only for CMake < 3.16
- one only for C++ < 14
- one only for boost < 1.71

**Pull request closed without merging**